src/test/java/com/google/research/bleth/services/DatabaseServiceTest.java

```diff
@@ -0,0 +1,264 @@
+package com.google.research.bleth.services;
+
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.dev.LocalDatastoreService;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.research.bleth.simulator.Agent;
+import com.google.research.bleth.simulator.Board;
+import com.google.research.bleth.simulator.Location;
+import junit.framework.TestCase;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+@RunWith(MockitoJUnitRunner.class)
+public final class DatabaseServiceTest extends TestCase {
+
+    private final LocalServiceTestHelper helper =
+            new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig()
+                    .setAutoIdAllocationPolicy(LocalDatastoreService.AutoIdAllocationPolicy.SEQUENTIAL));
+
+    private static final String firstSimulationId = "test-tracing-sim-id-1";
+    private static final String secondSimulationId = "test-tracing-sim-id-2";
+    private static final int roundOne = 1;
+    private static final int roundTwo = 2;
+
+    private static final Location zeroOnZeroCoordinate = new Location(0, 0);
+    private static final Location oneOnOneCoordinate = new Location(1, 1);
+    private static final Location zeroOnOneCoordinate = new Location(0, 1);
+    private static final Location oneOnZeroCoordinate = new Location(1, 0);
+
+    @Mock
+    private Agent firstAgent;
+    @Mock
+    private Agent secondAgent;
+
+    @Before
+    public void setUp() {
+        helper.setUp();
+    }
+
+    @Test
+    public void writeAndThenReadEmptyRealBoardState_ShouldGetEqualBoardState() {
+
+        DatabaseService db = DatabaseService.getInstance();
+        String emptyRealBoardState = new Board(2, 2).getState();
+
+        db.writeRealBoardState(firstSimulationId, roundOne, emptyRealBoardState);
+        String outputRealBoardState = db.getRealBoardState(firstSimulationId, roundOne);
+
+        assertThat(emptyRealBoardState).isEqualTo(outputRealBoardState);
+    }
+
+    @Test
+    public void writeAndThenReadNonRealBoardState_ShouldGetEqualBoardState() {
+
+        DatabaseService db = DatabaseService.getInstance();
+        Board nonEmptyRealBoard = new Board(2, 2);
+        Mockito.when(firstAgent.moveTo()).thenReturn(zeroOnZeroCoordinate);
+        nonEmptyRealBoard.placeAgent(firstAgent.moveTo(), firstAgent);
```

Comment:
You can just place firstBeacon at the wanted location on the board with placeAgent, there's no need to use Mockito or moveTo for that.